src/lib/viz/style/helpers/color-categories-style.ts

```diff
@@ -53,27 +54,69 @@ export function colorCategoriesStyle(
 
     validateParameters(opts);
 
-    let categories;
+    const categories = getCategories(opts, meta, featureProperty).slice(0, opts.top);
 
-    if (opts.categories.length) {
-      categories = opts.categories;
-    } else {
-      const stats = meta.stats.find(c => c.name === featureProperty) as CategoryFieldStats;
+    return calculateWithCategories(featureProperty, categories, meta.geometryType, opts);
```

Comment:
I'm gonna propose a refactor for a second phase on style helpers (not now).

Now that we have also 'legends' code inside the helpers we might need to rename certain functions... We might even.consider falling back into classes + factory methods, instead of keeping the pure functions... we'll see 😄